paddle/fluid/operators/mkldnn/activation_mkldnn_op.cc

```diff
@@ -96,10 +95,10 @@ void eltwise_forward(const framework::ExecutionContext &ctx,
       x->dims().size() == 2 || x->dims().size() == 3 || x->dims().size() == 4,
       "Input dim must be with 2, 3 or 4");
 
-  std::vector<int> src_tz = framework::vectorize2int(x->dims());
+  std::vector<int64_t> src_tz = paddle::framework::vectorize(x->dims());
```

Comment:
How about?
```
auto src_tz = paddle::framework::vectorize(x->dims());
```
Maybe we can unify `vectorize` and `vectorize2int` at first.

Reply:
Code You proposed does work for v1.0 but will crash in v0.2 because vectorize() returns int64_t vector and mkldnn v0.2 kernel expects int vector. We can't unify this as cuda kernels also use vectorize2int() function and vectorize() is used in some other places not maintained by us. Improvents like using auto will be made during code cleanup and refactor in the near future.

Reply:
What if we have template parametrised the vectorize function, so that by default it returns int64_t but it can be chosen through a template parameter what data type we want our dimensions to be returned in?

Reply:
@Sand3r- I think it's OK. Could you create a separate PR to do this?

Reply:
I'll do that

Reply:
Done in https://github.com/PaddlePaddle/Paddle/pull/19611